core/errors/error-match-error.ts

```diff
@@ -2,7 +2,7 @@ import { MatchError } from "./match-error";
 
 export class ErrorMatchError extends MatchError {
 
-   public constructor(actualError: Error,
+   public constructor(actualError: Error | undefined | null,
```

Comment:
I had to  add both because both assumptions were made, I'll try for just null. Do you want to use a union type, `Error | null`, or a type alias, say `type NullableError = Error | null`?